From e50daaf569089eaf86e7b618fc1ba1e3716340a0 Mon Sep 17 00:00:00 2001
From: george <george@gvlt>
Date: Wed, 28 Sep 2016 14:40:40 -0700
Subject: [PATCH 1/2] AutoTypePlatformX11::SendKeyPressedEvent() in order to
 prevent missing keys on some websites (likely due to Javascript on pages)
 Would be better if this was in config... works for now.

---
 src/autotype/xcb/AutoTypeXCB.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/autotype/xcb/AutoTypeXCB.cpp b/src/autotype/xcb/AutoTypeXCB.cpp
index becb65e0..856f2d6d 100644
--- a/src/autotype/xcb/AutoTypeXCB.cpp
+++ b/src/autotype/xcb/AutoTypeXCB.cpp
@@ -808,6 +808,11 @@ void AutoTypePlatformX11::SendKeyPressedEvent(KeySym keysym)
 
     /* restore the old keyboard mask */
     SendModifier(&event, release_mask, KeyPress);
+	
+	timespec ts;
+    ts.tv_sec = 0;
+    ts.tv_nsec = 30 * 1000 * 1000;
+    nanosleep(&ts, nullptr);
 }
 
 int AutoTypePlatformX11::MyErrorHandler(Display* my_dpy, XErrorEvent* event)

From 354290a65f245bfea7afdad7da5d765a04ba1134 Mon Sep 17 00:00:00 2001
From: george <george@gvlt>
Date: Wed, 28 Sep 2016 17:05:40 -0700
Subject: [PATCH 2/2] Moved AutoTypeInterCharacterDelay to AutoType.cpp /
 Config

---
 src/autotype/AutoType.cpp        | 8 ++++++--
 src/autotype/xcb/AutoTypeXCB.cpp | 5 -----
 src/core/Config.cpp              | 1 +
 3 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/autotype/AutoType.cpp b/src/autotype/AutoType.cpp
index 8032c27d..b047a506 100644
--- a/src/autotype/AutoType.cpp
+++ b/src/autotype/AutoType.cpp
@@ -338,7 +338,7 @@ QList<AutoTypeAction*> AutoType::createActionFromTemplate(const QString& tmpl, c
     QString tmplName = tmpl.toLower();
     int num = -1;
     QList<AutoTypeAction*> list;
-
+	
     QRegExp repeatRegEx("(.+) (\\d+)", Qt::CaseSensitive, QRegExp::RegExp2);
     if (repeatRegEx.exactMatch(tmplName)) {
         tmplName = repeatRegEx.cap(1);
@@ -357,7 +357,7 @@ QList<AutoTypeAction*> AutoType::createActionFromTemplate(const QString& tmpl, c
             return list;
         }
     }
-
+	
     if (tmplName == "tab") {
         list.append(new AutoTypeKey(Qt::Key_Tab));
     }
@@ -495,6 +495,10 @@ QList<AutoTypeAction*> AutoType::createActionFromTemplate(const QString& tmpl, c
             }
             else {
                 list.append(new AutoTypeChar(ch));
+				if (config()->get("AutoTypeInterKeyDelay").toInt() > 0)
+				{
+					list.append(new AutoTypeDelay(config()->get("AutoTypeInterKeyDelay").toInt()));
+				}
             }
         }
     }
diff --git a/src/autotype/xcb/AutoTypeXCB.cpp b/src/autotype/xcb/AutoTypeXCB.cpp
index 856f2d6d..becb65e0 100644
--- a/src/autotype/xcb/AutoTypeXCB.cpp
+++ b/src/autotype/xcb/AutoTypeXCB.cpp
@@ -808,11 +808,6 @@ void AutoTypePlatformX11::SendKeyPressedEvent(KeySym keysym)
 
     /* restore the old keyboard mask */
     SendModifier(&event, release_mask, KeyPress);
-	
-	timespec ts;
-    ts.tv_sec = 0;
-    ts.tv_nsec = 30 * 1000 * 1000;
-    nanosleep(&ts, nullptr);
 }
 
 int AutoTypePlatformX11::MyErrorHandler(Display* my_dpy, XErrorEvent* event)
diff --git a/src/core/Config.cpp b/src/core/Config.cpp
index 5b06c22d..9dc7d2e8 100644
--- a/src/core/Config.cpp
+++ b/src/core/Config.cpp
@@ -97,6 +97,7 @@ void Config::init(const QString& fileName)
     m_defaults.insert("MinimizeOnCopy", false);
     m_defaults.insert("UseGroupIconOnEntryCreation", false);
     m_defaults.insert("AutoTypeEntryTitleMatch", true);
+	m_defaults.insert("AutoTypeInterKeyDelay", 10);
     m_defaults.insert("security/clearclipboard", true);
     m_defaults.insert("security/clearclipboardtimeout", 10);
     m_defaults.insert("security/lockdatabaseidle", false);
